libraries/core/src/test/java/com/paritytrading/philadelphia/FIXInitiatorTest.java

```diff
@@ -61,10 +63,13 @@ void setUp() throws IOException {
         initiatorMessages = new FIXMessages();
         acceptorMessages  = new TestMessages();
 
-        initiatorStatus = new FIXConnectionStatus<>();
+        initiatorStatus = new FIXConnectionStatus();
 
         Channels channels = createChannels();
-        initiator = new FIXConnection<>(channels.initiator, initiatorConfig, initiatorMessages, initiatorStatus);
+        initiator = new FIXConnection(
+                channels.initiatorRead, channels.initiatorWrite,
```

Comment:
With this change, separating the roles, we won't need the equivalent of out `PipeChannel` testing class if we want to use pipes in PROD code - we can use the upstream and downstream pipes directly.